src/seedu/addressbook/data/person/Email.java

```diff
@@ -3,59 +3,63 @@
-
-    public static final String EXAMPLE = "valid@e.mail";
-    public static final String MESSAGE_EMAIL_CONSTRAINTS =
-            "Person emails should be 2 alphanumeric/period strings separated by '@'";
-    public static final String EMAIL_VALIDATION_REGEX = "[\\w\\.]+@[\\w\\.]+";
-
-    public final String value;
-    private boolean isPrivate;
-
-    /**
-     * Validates given email.
-     *
-     * @throws IllegalValueException if given email address string is invalid.
-     */
-    public Email(String email, boolean isPrivate) throws IllegalValueException {
-        this.isPrivate = isPrivate;
-        email = email.trim();
-        if (!isValidEmail(email)) {
-            throw new IllegalValueException(MESSAGE_EMAIL_CONSTRAINTS);
-        }
-        this.value = email;
-    }
-
-    /**
-     * Checks if a given string is a valid person email.
-     */
-    public static boolean isValidEmail(String test) {
-        return test.matches(EMAIL_VALIDATION_REGEX);
-    }
-
-    @Override
-    public String toString() {
-        return value;
-    }
-
-    @Override
-    public boolean equals(Object other) {
-        return other == this // short circuit if same object
-                || (other instanceof Email // instanceof handles nulls
-                && this.value.equals(((Email) other).value)); // state check
-    }
-
-    @Override
-    public int hashCode() {
-        return value.hashCode();
-    }
-
-
-    public boolean isPrivate() {
-        return isPrivate;
-    }
+public class Email implements Printable {
+
+	public static final String EXAMPLE = "valid@e.mail";
+	public static final String MESSAGE_EMAIL_CONSTRAINTS = "Person emails should be 2 alphanumeric/period strings separated by '@'";
+	public static final String EMAIL_VALIDATION_REGEX = "[\\w\\.]+@[\\w\\.]+";
+
+	public final String value;
+	private boolean isPrivate;
+
+	/**
+	 * Validates given email.
+	 *
+	 * @throws IllegalValueException
+	 *             if given email address string is invalid.
+	 */
+	public Email(String email, boolean isPrivate) throws IllegalValueException {
+		this.isPrivate = isPrivate;
+		email = email.trim();
+		if (!isValidEmail(email)) {
+			throw new IllegalValueException(MESSAGE_EMAIL_CONSTRAINTS);
+		}
+		this.value = email;
+	}
+
+	/**
+	 * Checks if a given string is a valid person email.
+	 */
+	public static boolean isValidEmail(String test) {
+		return test.matches(EMAIL_VALIDATION_REGEX);
+	}
+
+	@Override
+	public String toString() {
+		return value;
+	}
+
+	@Override
+	public boolean equals(Object other) {
+		return other == this // short circuit if same object
+				|| (other instanceof Email // instanceof handles nulls
+						&& this.value.equals(((Email) other).value)); // state
+																		// check
+	}
+
+	@Override
+	public int hashCode() {
+		return value.hashCode();
+	}
+
+	public boolean isPrivate() {
+		return isPrivate;
+	}
+
+	public String getPrintableString() {
+		return "Email: " + value;
```

Comment:
Is the indentation off? 
Is that why the whole class appears as a new commit? 
Please check and avoid it.